examples/flutter_gallery/lib/demo/material/full_screen_dialog_demo.dart

```diff
@@ -164,19 +165,41 @@ class FullScreenDialogDemoState extends State<FullScreenDialogDemo> {
           children: <Widget>[
             new Container(
               padding: const EdgeInsets.symmetric(vertical: 8.0),
-              decoration: new BoxDecoration(
-                border: new Border(bottom: new BorderSide(color: theme.dividerColor))
-              ),
               alignment: Alignment.bottomLeft,
-              child: new Text('Event name', style: theme.textTheme.display2)
+              child: new TextField(
+                decoration: const InputDecoration(
+                  labelText: 'Event name',
+                  hintText: 'What is the name of the event?',
+                  filled: true
+                ),
+                style: theme.textTheme.headline,
+                onChanged: (String value) {
+                  setState(() {
+                    if (value.isNotEmpty) {
+                      _eventName = value;
+                    } else {
+                      _eventName = 'New event';
+                    }
+                    _saveNeeded = true;
+                  });
+                }
+              )
             ),
             new Container(
               padding: const EdgeInsets.symmetric(vertical: 8.0),
-              decoration: new BoxDecoration(
-                border: new Border(bottom: new BorderSide(color: theme.dividerColor))
-              ),
               alignment: Alignment.bottomLeft,
-              child: new Text('Location', style: theme.textTheme.title.copyWith(color: Colors.black54))
+              child: new TextField(
+                decoration: const InputDecoration(
+                  labelText: 'Location',
+                  hintText: 'Where is the event?',
+                  filled: true
+                ),
+                onChanged: (String value) {
+                  setState(() {
+                    _saveNeeded = true;
```

Comment:
Only set _saveNeeded when the location is actually set. Need a _location state variable